src/sentry/tasks/relay.py

```diff
@@ -154,3 +108,162 @@ def schedule_update_config_cache(
     # To avoid that, make sure we first schedule the task, and only then mark
     # the project as debounced.
     projectconfig_debounce_cache.debounce(public_key, project_id, organization_id)
+
+
+def validate_args(organization_id=None, project_id=None, public_key=None):
+    """Validates arguments for the tasks and sets sentry scope.
+
+    The tasks should be invoked for only one of these arguments, however because of Celery
+    we want to use primitive types for the arguments.  This is the common validation to make
+    sure only one is provided.
+    """
+    if [bool(organization_id), bool(project_id), bool(public_key)].count(True) != 1:
+        raise TypeError("Must provide exactly one of organzation_id, project_id or public_key")
+
+    if project_id:
+        set_current_event_project(project_id)
+    if organization_id:
+        # Cannot use bind_organization_context here because we do not have a
+        # model and don't want to fetch one
+        sentry_sdk.set_tag("organization_id", organization_id)
+    if public_key:
+        sentry_sdk.set_tag("public_key", public_key)
+
+
+def project_keys_to_update(organization_id=None, project_id=None, public_key=None):
+    """Returns the project keys which need to have their config updated.
+
+    Queries the database for the required project keys.
+    """
+    from sentry.models import Project, ProjectKey
+
+    if organization_id:
+        projects = list(Project.objects.filter(organization_id=organization_id))
+        keys = list(ProjectKey.objects.filter(project__in=projects))
+    elif project_id:
+        projects = [Project.objects.get(id=project_id)]
+        keys = list(ProjectKey.objects.filter(project__in=projects))
+    elif public_key:
+        try:
+            keys = [ProjectKey.objects.get(public_key=public_key)]
+        except ProjectKey.DoesNotExist:
+            # In this particular case, where a project key got deleted and
+            # triggered an update, we know that key doesn't exist and we want to
+            # avoid creating more tasks for it.
+            #
+            # In other similar cases, like an org being deleted, we potentially
+            # cannot find any keys anymore, so we don't know which cache keys
+            # to delete.
+            projectconfig_cache.set_many({public_key: {"disabled": True}})
```

Comment:
nit: setting disabled keys in the cache doesn't really fit in this function, I think it should happen in the callers instead.

Reply:
hum, it really is something that should be handled by the tasks.  i guess you could argue something outside of the task should be setting this cache entry, but that seems a change that is rather risky for no gain here.